crates/bevy_asset/src/assets.rs

```diff
@@ -137,10 +137,16 @@ impl<T: Asset> Assets<T> {
         borrowed
     }
 
+    /// An iterator visiting all key-value pairs in arbitrary order.
     pub fn iter(&self) -> impl Iterator<Item = (HandleId, &T)> {
         self.assets.iter().map(|(k, v)| (*k, v))
     }
 
+    /// An iterator visiting all key-value pairs in arbitrary order, with mutable references to the values.
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = (HandleId, &mut T)> {
```

Comment:
We can't just return mutable references here because this won't produce mutation events like `set` does. If we are going to expose this api, we need to send a mutation event for every item in the collection.

Reply:
Ah ok, makes sense.